<!DOCTYPE html>
<html>
  <head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <meta name="description" content="">
    <meta name="author" content="">
    <title>MiTranscriptome</title>
    <link rel="shortcut icon" type="image/x-icon" href="{{ url_for('static', filename='images/block_m_icon3.png') }}">
    <!-- Bootstrap CSS -->
    <link href="{{ url_for('static', filename='bootstrap/css/bootstrap-yeti.min.css') }}" rel="stylesheet">

    <!-- DataTables CSS -->    
    <link href="{{ url_for('static', filename='datatables/css/dataTables.bootstrap.css') }}" rel="stylesheet">
    <!-- DataTables Custom CSS -->    
    <link href="{{ url_for('static', filename='css/datatables.custom.css') }}" rel="stylesheet">

    <!-- Custom CSS for this template-->
    <link href="{{ url_for('static', filename='css/custom.css') }}" rel="stylesheet">

     <!-- data-main attribute tells require.js to load
          scripts/main.js after require.js loads. -->
     <script data-main="{{ url_for('static', filename='scripts/main') }}" src="{{ url_for('static', filename='scripts/libs/require.js') }}"></script>

  </head>
  <body>

    <!-- Fixed horizontal navbar -->
    <div class="navbar navbar-default navbar-fixed-top" role="navigation">
      <div class="container-fluid">
		    <div class="navbar-header">
          <a id="btn-home" href="#" class="navbar-brand"><span><img src="{{ url_for('static', filename='images/block_m_white.png') }}" height="20" width="28"></span>&nbsp;MiTranscriptome <i>(alpha)</i></a>
		    </div>
				<div class="collapse navbar-collapse">
				  <ul class="nav navbar-nav navbar-right">
		      	<li class="dropdown">
		          <a class="dropdown-toggle" data-toggle="dropdown">UCSC Browser Links <b class="caret"></b></a>
		          <ul class="dropdown-menu">
				        <li><a href="http://genome.ucsc.edu/cgi-bin/hgTracks?hgS_doOtherUser=submit&hgS_otherUserName=mitranscriptome&hgS_otherUserSessionName=mitranscriptome" 
			               target = "_blank" >View MiTranscriptome lncRNAs</a></li>
			          <li><a href="http://genome.ucsc.edu/cgi-bin/hgTracks?hgS_doOtherUser=submit&hgS_otherUserName=mitranscriptome&hgS_otherUserSessionName=mitranscriptome_all" 
			               target = "_blank" >View Entire Assembly</a></li>
			        </ul>
			      </li>    
	          <li><a href="#" data-toggle="modal" data-target="#modal-about">About</a></li>
	          <li><a href="#" data-toggle="modal" data-target="#modal-help">Help</a></li>
	        </ul>
	      </div>
	      </div><!--/.container -->
    </div><!--/.navbar -->

    <div class="modal fade" id="modal-about" tabindex="-1" role="dialog" aria-labelledby="modal-label-about" aria-hidden="true">
      <div class="modal-dialog">
        <div class="modal-content">
          <div class="modal-header">
            <button type="button" class="close" data-dismiss="modal" aria-hidden="true">&times;</button>
            <h4 class="modal-title" id="modal-label-about">About</h4>
          </div>
          <div class="modal-body">
            <p>MiTranscriptome is a large-scale <i>ab initio</i> transcriptome 
            meta-assembly from 6,503 RNA-Seq libraries. RNA-Seq data were obtained
            from <a target = "_blank" href="http://cancergenome.nih.gov">The Cancer Genome Atlas (TCGA)</a>,
            the <a target = "_blank" href="http://www.genome.gov/encode">ENCODE Project</a>, the 
            <a target = "_blank" href="http://mctp.med.umich.edu">Michigan Center for Translational Pathology</a>,
            and other published sources. The lncRNAs most differentially expressed in cancers 
            and tissues are reported in this portal.</p>
            <p><h5>Methods</h5>
            <p>Large-scale transcriptome meta-assembly was performed using AssemblyLine, which
            rigorously filters background noise and artifacts before employing a greedy
            dynamic programming approach to derive a consensus transcriptome from 
            individual samples.</p>
            <p>Expression analysis was performed using a novel non-parametric tool 
            called Sample Set Enrichment Analysis (SSEA), which uses weighted 
            Kolmorgorov-Smirnoff-like tests to detect cancer-specific and lineage-specific
            patterns. SSEA adapts ideas from 
            <a target = "_blank" href="http://www.broadinstitute.org/gsea/index.jsp">Gene Set Enrichment Analysis</a>.</p>
            <p><h5>Source Code:</h5>
            <ul>   
              <li><a href="https://code.google.com/p/ssea" target = "_blank">SSEA Source Code</a></li>
              <li><a href="https://code.google.com/p/assemblyline" target = "_blank">AssemblyLine Source Code</a></li>
            </ul>
            <p><h5>Contributors</h5>
            <ul>
              <li>Matthew K. Iyer (MD/PhD Student)</li>
              <li>Yashar S. Niknafs (MD/PhD Student)</li>
              <li>Arul M. Chinnaiyan, M.D., Ph.D. (Principal Investigator)</li>
            </ul></p>
            <p><h5>Acknowledgements</h5>
            <p><ul>
              <li>Anirban Sahu (Web Design)</li>
              <li>Terry Barrette (System Administration)</li>
            </ul></p>
            <p><h6>
            MiTranscriptome was developed by members of <a target = "_blank" href="http://mctp.med.umich.edu">Michigan Center for 
            Translational Pathology</a> at the <a target = "_blank" href="http://www.umich.edu">University of Michigan</a>.
            </h6></p>
          </div>
          <div class="modal-footer">
            <button type="button" class="btn btn-default" data-dismiss="modal">Close</button>
          </div>
        </div><!-- /.modal-content -->
      </div><!-- /.modal-dialog -->
    </div><!-- /.modal -->

    <div class="modal fade" id="modal-help" tabindex="-1" role="dialog" aria-labelledby="modal-label-help" aria-hidden="true">
      <div class="modal-dialog">
        <div class="modal-content">
          <div class="modal-header">
            <button type="button" class="close" data-dismiss="modal" aria-hidden="true">&times;</button>
            <h4 class="modal-title" id=""modal-label-help"">Help</h4>
          </div>
          <div class="modal-body">
            <p><h4>MiTranscriptome is undergoing active development. Please report issues and recommend 
            features here:
            <a target = "_blank" href="https://groups.google.com/d/forum/mitranscriptome">User Group</a>.</p></h4>
          </div>
          <div class="modal-footer">
            <button type="button" class="btn btn-default" data-dismiss="modal">Close</button>
          </div>
        </div><!-- /.modal-content -->
      </div><!-- /.modal-dialog -->
    </div><!-- /.modal -->    
    
    <div class="container-fluid">
      <div id="div-welcome" class="jumbotron">
        <h2>Welcome to the MiTranscriptome lncRNA Portal</h2>
        <p><small>MiTranscriptome is <i>ab initio</i> transcriptome 
        reconstruction from 6,503 RNA-seq libraries spanning diverse 
        cancer and tissue types. The MiTranscriptome contains >58,000 
        lncRNAs, the vast majority of which are currently unannotated. 
        This portal reports the long non-coding RNAs (lncRNAs) that are expressed in a 
        lineage-specific or cancer-specific manner.</small></p>
        <br>
        <p id='wait-msg'>Please wait while the transcript browser loads...</p>
        <p><a id='btn-get-started' class="toshow btn btn-lg btn-primary" href="#" role="button">Get Started &raquo;</a></p>
		    <div class="row text-right">
		  	  <img src="{{ url_for('static', filename='images/chrome.png') }}" height="16" width="16">
		  		<img src="{{ url_for('static', filename='images/firefox.png') }}" height="16" width="16">
		  		<img src="{{ url_for('static', filename='images/safari.png') }}" height="16" width="16">
		  		<img src="{{ url_for('static', filename='images/no-ie.png') }}" height="16" width="16">
		  	</div>
      </div><!-- /div-welcome -->

	    <!-- Transcript Table -->
	    <div id="div-transcript-browser" class="toshow container-fluid">
	      <table id="table-transcripts" class="table table-condensed table-striped text-center">
	      <!-- <caption>Transcript Browser</caption> -->
			  <thead>
				<tr>
				<th></th>
		    <th>Name</th>
		    <th><a href="#" data-toggle="tooltip" data-html="true" title="UCSC GRCh37/hg19">Location</a></th>
		    <th><a href="#" data-toggle="tooltip" data-html="true" title="Considered annotated if transcript has same-strand exonic overlap with UCSC, RefSeq, or GENCODE 19">Annotated?</a></th>
		    <th><a href="#" data-toggle="tooltip" data-html="true" title="Intergenic if transcript has no overlap with any annotated transcripts or their introns">Intergenic?</a></th>
		    <th><a href="#" data-toggle="tooltip" data-html="true" title="Transcripts of Unknown Coding Potential (TUCPs) possess putative coding regions">Type</a></th>
		    <th><a href="#" data-toggle="tooltip" data-html="true" title="Highly Conserved lncRNAs (HICLNCs) possess ultraconserved elements">HICLNC?</a></th>
		    <th><a href="#" data-toggle="tooltip" data-html="true" 
		         title="Cancer-Specific&nbsp;<span class='label label-success'>C</span>
		         <br/>Lineage-Specific&nbsp;<span class='label label-warning'>L</span>">
		         Association
		         </a>
		    </th>
		    <th><a href="#" data-toggle="tooltip" data-html="true" title="SSEA analysis performed on 27 tissue types">Tissue Type</a></th>
		    <th><a href="#" data-toggle="tooltip" data-html="true"
		         title="<span class='glyphicon glyphicon-arrow-up red'/>Over-expression
		         <br/><span class='glyphicon glyphicon-arrow-down blue'/>Under-expression">
		         Percentile
		         </a>
		    </th>
		    <th><a href="#" data-toggle="tooltip" data-html="true" title="Fragments per kilobase per million (FPKM)">Expression Mean</a></th>
		    <th><a href="#" data-toggle="tooltip" data-html="true" title="Fragments per kilobase per million (FPKM)">Expression 95%</a></th>
		    <th><a href="#" data-toggle="tooltip" data-html="true" title="Fragments per kilobase per million (FPKM)">Expression 99%</a></th>
				</tr>
			  </thead>
			  <tbody>
			  </tbody>
			  </table>
			</div><!-- /div-table -->

    </div><!-- /container -->

  </body>
</html>        


