<!DOCTYPE html>
<html>
  <head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <meta name="description" content="">
    <meta name="author" content="">
    <link rel="shortcut icon" href="{{ url_for('static', filename='BlockM.ico') }}">
    <title>MiTranscriptome</title>

    <!-- Bootstrap CSS -->
    <link href="{{ url_for('static', filename='bootstrap/css/bootstrap.css') }}" rel="stylesheet">
    <link href="{{ url_for('static', filename='bootstrap/css/bootstrap-responsive.css') }}" rel="stylesheet">

    <!-- Selectize CSS -->
    <link href="{{ url_for('static', filename='css/selectize.css') }}" rel="stylesheet">
    <link href="{{ url_for('static', filename='css/selectize.bootstrap3.css') }}" rel="stylesheet">

    <!-- tablesorter bootstrap widget theme -->
    <link href="{{ url_for('static', filename='css/tablesorter.theme.bootstrap.css') }}" rel="stylesheet">

    <!-- Custom CSS for this template-->
    <link href="{{ url_for('static', filename='css/navbar-fixed-top.css') }}" rel="stylesheet">
    <link href="{{ url_for('static', filename='css/custom.css') }}" rel="stylesheet">

     <!-- data-main attribute tells require.js to load
          scripts/main.js after require.js loads. -->
     <script data-main="{{ url_for('static', filename='scripts/main') }}" src="{{ url_for('static', filename='scripts/libs/require.js') }}"></script>
     
     <!-- anirban toggle functions -->
     <script src="{{ url_for('static', filename='scripts/custom.js') }}"></script>
  </head>
  <body>

    <!-- Fixed horizontal navbar -->
    <div class="navbar navbar-default navbar-fixed-top" role="navigation">
      <div class="container">
        <div class="navbar-header">
          <a href="/"><span class="glyphicon glyphicon-home"></span><h4> MiTranscriptome</h4></a>
        </div>
        <div class="navbar-collapse collapse">
          <ul class="nav navbar-nav navbar-right">
            <li><a href="http://genome.ucsc.edu/cgi-bin/hgTracks?hgS_doOtherUser=submit&hgS_otherUserName=mitranscriptome&hgS_otherUserSessionName=mitranscriptome" 
            target = "_blank" >View lncRNAs in UCSC Browser</a></li>
            <li><a href="http://genome.ucsc.edu/cgi-bin/hgTracks?hgS_doOtherUser=submit&hgS_otherUserName=mitranscriptome&hgS_otherUserSessionName=mitranscriptome_all" 
            target = "_blank" >View Assembly in UCSC Browser</a></li>
            <li><a href="#" data-toggle="modal" data-target="#myModal_About">About</a></li>
          </ul>
        </div>
      </div><!--/.container -->
    </div><!--/.navbar -->

    <div class="modal fade" id="myModal_About" tabindex="-1" role="dialog" aria-labelledby="myModalLabel_About" aria-hidden="true">
      <div class="modal-dialog">
        <div class="modal-content">
          <div class="modal-header">
            <button type="button" class="close" data-dismiss="modal" aria-hidden="true">&times;</button>
            <h4 class="modal-title" id="myModalLabel_About">About</h4>
          </div>
          <div class="modal-body">
            <p><h4>MiTranscriptome is currently in beta testing. Please report issues at our
            <a target = "_blank" href="https://github.com/mkiyer/mitranscriptome/issues">github site</a>.</h4>
            </p>
            <p>MiTranscriptome is a large-scale <i>ab initio</i> transcriptome 
            meta-assembly from 6,503 RNA-Seq libraries. RNA-Seq data were obtained
            from <a target = "_blank" href="http://cancergenome.nih.gov">The Cancer Genome Atlas (TCGA)</a>,
            the <a target = "_blank" href="http://www.genome.gov/encode">ENCODE Project</a>, the 
            <a target = "_blank" href="http://mctp.med.umich.edu">Michigan Center for Translational Pathology</a>,
            and other published sources. The lncRNAs most differentially expressed in cancers 
            and tissues are reported in this portal.</p>
            <p><h5>Methods</h5>
            <p>Large-scale transcriptome meta-assembly was performed using AssemblyLine, which
            rigorously filters background noise and artifacts before employing a greedy
            dynamic programming approach to derive a consensus transcriptome from 
            individual samples.</p>
            <p>Expression analysis was performed using a novel non-parametric tool 
            called Sample Set Enrichment Analysis (SSEA), which uses weighted 
            Kolmorgorov-Smirnoff-like tests to detect cancer-specific and lineage-specific
            patterns. SSEA adapts ideas from 
            <a target = "_blank" href="http://www.broadinstitute.org/gsea/index.jsp">Gene Set Enrichment Analysis</a>.</p>
            <p><h5>Contact:</h5>
            <ul>
              <li>Arul M. Chinnaiyan, M.D., Ph.D. (Principal Investigator): <i>arul@umich.edu</i></li>
              <li>Matthew K. Iyer (MD/PhD Student): <i>mkiyer@umich.edu</i></li>
              <li>Yashar S. Niknafs (MD/PhD Student): <i>yniknafs@umich.edu</i></li>
            </ul></p>
            <p><h5>Source Code:</h5>
            <ul>   
              <li><a href="https://code.google.com/p/ssea" target = "_blank">SSEA Source Code</a></li>
              <li><a href="https://code.google.com/p/assemblyline" target = "_blank">AssemblyLine Source Code</a></li>
            </ul>
            <p><h5>Acknowledgements:</h5>
            <p><ul>
              <li>Anirban Sahu (Web Design): <i>asahu@umich.edu</i></li>
              <li>Terry Barrette (System Administration): <i>trbarret@med.umich.edu</i></li>
            </ul></p>
            <p><h6>
            MiTranscriptome was developed by members of <a target = "_blank" href="http://mctp.med.umich.edu">Michigan Center for 
            Translational Pathology</a> at the <a target = "_blank" href="http://www.umich.edu">University of Michigan</a>.
            </h6></p>
          </div>
          <div class="modal-footer">
            <button type="button" class="btn btn-default" data-dismiss="modal">Close</button>
          </div>
        </div><!-- /.modal-content -->
      </div><!-- /.modal-dialog -->
    </div><!-- /.modal -->

    <!-- Main Component -->
    <div class="container" id="top">
    
        <div id="Home" class="jumbotron">
          <h2>Welcome to the MiTranscriptome lncRNA Portal</h2>
          <p><small>MiTranscriptome is <i>ab initio</i> transcriptome 
          reconstruction from 6,503 RNA-seq libraries spanning diverse 
          cancer and tissue types. The MiTranscriptome contains >58,000 
          lncRNAs, the vast majority of which are currently unannotated. 
          This portal reports the lncRNAs that are expressed in a 
          lineage-specific or cancer-specific manner.</small></p>
          <p><a class="btn btn-lg btn-primary" href="#" role="button" onclick="toggle_off('Home');toggle_on('Select_Transcripts')">Get Started &raquo;</a></p>
        </div><!-- /Home Screen -->

        <div id="Select_Transcripts" class="toshow jumbotron">
   		  <label for="select-transcripts"><h4>Select Tissue/Cancer Type:</h4></label>
   		  <select id="select-transcripts">
   		    <option value="">Cancer Type/Tissue Type</option>
            <option value="aml">Acute Myelogenous Leukemia (AMATs)</option>
            <option value="bladder">Bladder Cancer (BLCATs)</option>
            <option value="breast">Breast Cancer (BRCATs)</option>
            <option value="cervical">Cervical Cancer (CVATs)</option>
            <option value="cml">Chronic Myelogenous Leukemia (CMATs)</option>
            <option value="colorectal">Colorectal Cancer (CRATs)</option>
            <option value="gbm">Glioblastoma Multiforme (GBATs)</option>
            <option value="head_neck">Head and Neck Cancer (HNCATs)</option>
            <option value="heart">Heart Tissue (HRATs)</option>
            <option value="hesc">Human Embryonic Stem Cells (ESATs)</option>
            <option value="kich">Chromophobe Renal Cell Carcinoma (KCHCATs)</option>
            <option value="kirc">Renal Clear Cell Carcinoma (KCCATs)</option>
            <option value="kirp">Renal Papillary Cell Carcinoma (KPCATs)</option>
            <option value="lgg">Low Grade Glioma (LGATs)</option>
            <option value="liver">Liver Cancer (LVCATs)</option>
            <option value="luad">Lung Adenocarcinoma (LACATs)</option>
            <option value="lusc">Lung Squamous Cell Carcinoma (LSCATs)</option>
            <option value="medulloblastoma">Medulloblastoma (MBATs)</option>
            <option value="melanoma">Melanoma (MEATs)</option>
            <option value="mpn">Myeloproliferative Neoplasia (MPATs)</option>
            <option value="ovarian">Ovarian Cancer (OVATs)</option>
            <option value="pancreatic">Pancreatic Cancer (PNATs)</option>
            <option value="prostate">Prostate Cancer (PCATs)</option>
            <option value="skeletal_muscle">Skeletal Muscle Tissue (SMATs)</option>
            <option value="stomach">Stomach Cancer (STCATs)</option>
            <option value="thyroid">Thyroid Cancer (THCATs)</option>
            <option value="uterine">Uterine Endometrial Carcinoma (UTATs)</option>
            <option value="hiclinc">Highly Conserved Long Intergenic Non-Coding RNAs (HICLINCs)</option>
   		  </select>
        </div><!-- /Select Transcripts --> 

        <div id="div-selected-transcripts" class="toshow jumbotron">
          <a type="button" class="close" onclick="toggle_off('div-selected-transcripts');toggle_on('Select_Transcripts')">&times;</a>
          <div id="div-transcript-table"></div>
          <p></p>
        </div><!-- /Display table of transcripts -->

      </div><!-- /container -->
      
  </body>
</html>        


